### PR TITLE
Refactored Duration Data type

### LINKED ISSUE #394: Remove use of ISO 8601 in Duration to align with the current version of the standard
Revert duration values changes in #55 to ensure we can only use the values defined in [behaviour of time operators](https://sdmx-twg.github.io/vtl/2.1/html/reference_manual/typical_behaviour.html#behaviour-of-time-operators)

Image

As there is an inconsistency over the use of the Duration data type, we will use the Duration values specified and use String data types as arguments and result types of the Time Operators that use it

Check the following operators and related tests to ensure the Duration is applied correctly:

- The result of `period_indicator` has a Duration data type
- The input of `time_agg` supports both String and Duration
- The check_duration function checks that the values of the Duration data type are in the `PERIOD_INDICATOR_MAPPING` dict, and that those are the ones specified above
- The Duration data type comparison is perform adequately using the `PERIOD_INDICATOR_MAPPING` dict
- The return data type of the new operators `daytoyear` and `daytomonth` is String
- The input data type of the new operators `yeartoday` and `monthtoday` is String, a regex will be used to ensure it matches an ISO 8601 Duration

Relevant operators links on VTL docs:
- [Days to duration](https://sdmx-twg.github.io/vtl/2.1/html/reference_manual/operators/Time%20operators/Number%20days%20to%20duration/index.html#)
- [Duration to number](https://sdmx-twg.github.io/vtl/2.1/html/reference_manual/operators/Time%20operators/Duration%20to%20number%20days/index.html)
- [Period indicator](https://sdmx-twg.github.io/vtl/2.1/html/reference_manual/operators/Time%20operators/Period%20indicator/index.html)

### PR DESCRIPTION
**Pull request by albertohernandez1995**
## Major Changes
- Refactored Duration Data Type by rollbacking it to the previous state, without the ISO 8601 check.
- Refactored related test.
- Moved to_days method to Unary main Datatype to keep supporting the ISO 8601 check at new time unary operators.
- Refactored new time operators as requested in the issue.

> [!NOTE]
>  When Duration data type is clearly defined, we should reuse the comment exception tests in typechecking / string module. They don't raise up right now because they are not longer exceptions and its values are in fact Duration values.

---
closes #394

### CODE REVIEW

File: src/vtlengine/Operators/__init__.py [hunk 1]
```diff
@@ -898,3 +899,24 @@ def component_evaluation(cls, operand: DataComponent) -> DataComponent:
             operand.data.copy() if operand.data is not None else pd.Series()
         )
         return result_component
+
+    @classmethod
+    def to_days(cls, value: Any) -> int:
```
Comment (javihern98) [inline src/vtlengine/Operators/__init__.py]: Typing of value has to be str

**Review by javihern98 — COMMENTED** [review]
Small changes are needed, but overall looks good. I suggest to add as well tests using filter and a Scalar of type Duration with any of the Duration values and None (in test_additional_scalars)
